src/app.ts

```diff
@@ -1,24 +1,73 @@
-const Koa = require('koa');
-const cors = require('@koa/cors');
+#!/usr/bin/env node
 
-const { admin } = require('./firebaseAdmin');
-const { verifyFirebaseIdToken } = require('./verifyFirebaseIdToken');
+import path from 'path';
+
+import Koa from 'koa';
+import cors from '@koa/cors';
+import Router from 'koa-router';
+
+import { getAdmin } from './firebaseAdmin';
+import  { verifyFirebaseIdToken } from './verifyFirebaseIdToken';
+
+let admin;
+if (process.env.NODE_ENV === 'test' && process.argv.length > 2) {
+    const credentialsPathArg = process.argv[2];
+    if (path.isAbsolute(credentialsPathArg)) {
+        admin = getAdmin(credentialsPathArg);
+    } else {
+        // Resolve the path relative to the cwd.
+        admin = getAdmin(path.resolve(path.join(process.cwd(), credentialsPathArg)));
+    }
+} else {
+    admin = getAdmin();
+}
+
+const db = admin.firestore();
+const members = db.collection('members');
+const operations = db.collection('operations');
 
 const app = new Koa();
 
 app.use(cors());
 
-app.use(verifyFirebaseIdToken);
+// Put endpoints that don't need the user to be authenticated above this.
+app.use(verifyFirebaseIdToken(admin));
+// Put endpoints that do need the user to be authenticated below this.
 
-app.use(async ctx => {
-    // const db = admin.database();
-    // const members = db.collections('members');
-    ctx.body = ctx.url + ' ' + ctx.state.user; // + ' ' + members;
-});
+const authenticatedRouter = new Router()
+    .param('mid', async (mid, ctx, next) => {
+        const midQuery = (await members.where('mid', '==', mid).get());
+        if (midQuery.empty) {
+            ctx.status = 404;
+            return;
+        }
+        ctx.state.toMember = midQuery.docs[0];
+        return await next();
+    })
+    .post('/api/members/:mid/trust', async ctx => {
+        const authUid = ctx.state.user.uid;
+        const authMember = await members.doc(authUid).get();
+        try {
+            const newOperation = await operations.add({
```

Comment:
Can we
a) verify data is valid (that member creating operation really has joined etc), then
b) insert it with applied is true

or is that for separate PR?

Reply:
for a), do you mean that the member has been trusted? From the call today, it seems like we're okay with untrusted members performing operations? Maybe we need to think about that in more depth?
b) Removing applied all together.